lib/DxcSupport/dxcapi.extval.cpp

```diff
@@ -1,42 +1,463 @@
+// validation would normally be performed.
+class ExternalValidationCompiler : public IDxcCompiler2, public IDxcCompiler3 {
+private:
+  CComPtr<IDxcValidator> Validator;
+
+  // This wrapper wraps one particular compiler interface.
+  // When QueryInterface is called, we create a new wrapper
+  // for the requested interface, which wraps the result of QueryInterface
+  // on the compiler object. Compiler pointer is held as IUnknown and must
+  // be upcast to the appropriate interface on use.
+  IID CompilerIID;
+  CComPtr<IUnknown> Compiler;
+
+  // Cast current compiler interface pointer. Used from methods of the
+  // associated interface, assuming that the current compiler interface is
+  // correct for the method call.
+  // This will either be casting to the original interface retrieved by
+  // QueryInterface, or to one from which that interface derives.
+  template <typename T> T *castCompilerSafe() const {
+    // Compare stored IID with the IID of T
+    if (CompilerIID == __uuidof(T)) {
+      // Safe to cast because the underlying compiler object in
+      // Compiler originally implemented the interface T
+      return static_cast<T *>(Compiler.p);
+    }
+
+    return nullptr;
+  }
+
+  template <typename T> T *castCompilerUnsafe() {
+    if (T *Safe = castCompilerSafe<T>())
+      return Safe;
+    return static_cast<T *>(Compiler.p);
+  }
+
+public:
+  ExternalValidationCompiler(IMalloc *Malloc, IDxcValidator *OtherValidator,
+                             REFIID OtherCompilerIID, IUnknown *OtherCompiler)
+      : Validator(OtherValidator), CompilerIID(OtherCompilerIID),
+        Compiler(OtherCompiler), m_pMalloc(Malloc) {}
+
+  // IUnknown implementation
+  DXC_MICROCOM_TM_REF_FIELDS()
+  DXC_MICROCOM_TM_ADDREF_RELEASE_IMPL()
+  DXC_MICROCOM_TM_ALLOC(ExternalValidationCompiler)
+
+  // QueryInterface creates a new wrapper for the correct compiler interface
+  // retrieved by calling QueryInterface on the compiler. The new wrapper
+  // keeps the IID and returned compiler interface pointer for use in
+  // associated interface methods. Those methods can then use
+  // castCompilerUnsafe to upcast the pointer to the appropriate interface
+  // for the method.
+  HRESULT STDMETHODCALLTYPE QueryInterface(REFIID Iid,
+                                           void **ResultObject) override {
+    if (!ResultObject)
+      return E_POINTER;
+
+    *ResultObject = nullptr;
+
+    // Get pointer for an interface the compiler object supports.
+    CComPtr<IUnknown> TempCompiler;
+    IFR(Compiler->QueryInterface(Iid, (void **)&TempCompiler));
+
+    try {
+      // This can throw; do not leak C++ exceptions through COM method
+      // calls.
+      CComPtr<ExternalValidationCompiler> NewWrapper(
+          Alloc(m_pMalloc, Validator, Iid, TempCompiler));
+      return DoBasicQueryInterface<IDxcCompiler, IDxcCompiler2, IDxcCompiler3>(
+          NewWrapper.p, Iid, ResultObject);
```

Comment:
```suggestion
          NewWrapper.Get(), Iid, ResultObject);
```

Reply:
Doesn't seem like this is a valid use case for Get(), there are build errors using .Get(), it isn't a member function.